### PR TITLE
39 fix post sentence unknown language acceptance

### PR DESCRIPTION
**Pull request by jean553**

### CODE REVIEW

File: src/sentences.rs [hunk 1]
```diff
@@ -32,7 +36,7 @@ fn create_sentence<'r>(
         ) VALUES (
             $1,
             $2,
-            (SELECT id FROM language WHERE iso639_3 = $3)
+            COALESCE((SELECT id FROM language WHERE iso639_3 = $3), 0)
```
Comment (allan-simon) [inline src/sentences.rs]: you can put the comment inside the SQL

File: src/sentences.rs [hunk 2]
```diff
@@ -47,13 +51,16 @@ fn create_sentence<'r>(
...
-
-                return  Response::build()
+                return Response::build()
                     .status(Status::Conflict)
-                    .finalize()
-                ;
+                    .finalize();
             }
-            panic!(format!("{}", e));
+
```
Comment (allan-simon) [inline src/sentences.rs]: should specifically check the foreign key violatio

File: src/sentences.rs [hunk 3]
```diff
@@ -46,13 +55,19 @@ fn create_sentence<'r>(
...
-                return  Response::build()
+            let error = e.code();
+            if error == Some(&UNIQUE_VIOLATION) {
+                return Response::build()
                     .status(Status::Conflict)
-                    .finalize()
-                ;
+                    .finalize();
             }
+            else if error == Some(&FOREIGN_KEY_VIOLATION) {
```
Comment (allan-simon) [inline src/sentences.rs]: no need for a else as you have a return in the if

Comment (jean553) [reply src/sentences.rs]: You're right. Done.

**Pull request merged**